docs/decisions/0005-lti-pii-sharing-flag.rst

```diff
@@ -0,0 +1,101 @@
+called ``ask_to_send_username`` and ``ask_to_send_email`` that enable sending
+such PII to the external tool. However, `by default these options are not
+visible
+<https://github.com/edx/xblock-lti-consumer/blob/edec2a68282a2a1fc2b2036e1989d60688fa6b19/lti_consumer/lti_xblock.py#L588-L599>`_
+to course authors, and there is no way to send PII to LTI tools. In order to
+even enable these options to appear in the LTI XBlock configuration, you need
+to first set a `configuraiton flag
+<https://github.com/edx/edx-platform/blob/e19ba34f5a564285b3a20a7298c20ca640ca5aa0/cms/djangoapps/xblock_config/models.py#L38-L86>`_
+via Django admin in studio.
+
+This configuration flag unlocks the editing of the above fields after which
+those fields can be set to allow sharing of PII. The configuraiong flag works
+on a course-wide level, while the flags above work on a per-LTI-XBlock-basis.
+
+With the recent addition of LTI Course tabs, there is need for a similar
+mechanism for enabling sending of PII. What needs to be determined is whether
+we should use the same flag for the purpose, or create a new flag for this use
+with LTI tabs.
+
+The arguments for creating a new flag are:
+
+1. The scope of the current flag don't match the new use case.
+2. Currently the flag is implemented in studio, while the flag now needs to be
+   used in LMS and MFEs.
+3. Keeping flags separate allows controlling this flag separately for course
+   tabs vs XBlocks.
+
+The arguments for using the same flag are:
+
+1. As part of `EDUCATOR-121 <https://openedx.atlassian.net/browse/EDUCATOR-121>`_
+   this config flag is already supposed to be moved to the LTI XBlock, which
+   would make it usable from LMS and Studio.
+2. The name and scope can both be changed while retaining the data.
+3. It's confusing to have two flags with very similar purposes.
+4. There is already a double layer of flags here. Enabling this flag will enable
+   setting other flags, and adding another separate flag doesn't help. Even if
+   the flag is only needed for either tabs or XBlocks, there is another step
+   to enabling sharing PII which can simply be skipped if undesired.
+
+Decision
+========
+
+Given the above context it makes sense to move the existing flag to the LTI
+XBlock while making sure it uses the same data table. This move means that the
+flag will now be accessible via LMS and Studio, which is what is needed.
+
+The LTI course tabs code will be modified to check this value before allowing
+the sending of PII even if the tab's configuration allows sending PII.
+
+Currently, if this flag is enabled for a course, and the `ask_to_send_*` flags
+are enabled for an XBlock, disabling this flag will not stop the XBlock from
+sending PII. This flag will *only* disable editing the `ask_to_send_*` flags.
+
+This might not be desirable behaviour. So we can also modify the scope of this
+flag to be the final arbiter of whether PII sharing is allowed or not instead
```

Comment:
In addition to this, it would be user-friendly to show the course authors that there is such a configuration flag, which is currently disabled for a course, and that's why these fields are not being displayed. What do you think?

Reply:
I've currently based it on the way the XBloock work, which is to just hide the fields. Since they are hidden by default, only the people who need to be aware of them will ever see them.

Reply:
I agree that we should improve the user-friendliness here.  It makes sense to me that the flag would be the final arbiter/override the ask_to_send_* settings, and it may also be worth giving users a note that says how they can get it enabled if necessary.

Reply:
Sure I've updated the doc to include this. Would this apply to the XBlock as well? Or just new UI for the MFEs?